File: python/tracing/anthropic-agents-sdk/basic/wrapped_query_test.py
"""
Wrapped Query — the simplest integration pattern.

Uses exporter.query() which handles hooks + tracking automatically.
One line to instrument, zero boilerplate.

Setup:
    pip install claude-agent-sdk respan-exporter-anthropic-agents python-dotenv

Run:
    python basic/wrapped_query_test.py

    # or with pytest:
    pytest basic/wrapped_query_test.py -v
"""

from dotenv import load_dotenv

load_dotenv(override=True)

import asyncio
import os

import pytest
from claude_agent_sdk import ClaudeAgentOptions

from respan_exporter_anthropic_agents import RespanAnthropicAgentsExporter
from _sdk_runtime import query_for_result

API_KEY = os.getenv("RESPAN_API_KEY")
BASE_URL = os.getenv("RESPAN_BASE_URL")

exporter = RespanAnthropicAgentsExporter(
    api_key=API_KEY,
    base_url=BASE_URL,
)


@pytest.mark.asyncio
async def test_wrapped_query():
    """Use exporter.query() for automatic tracing — simplest pattern."""

    message_types = []

    def _on_message(message):
        msg_type = type(message).__name__
        message_types.append(msg_type)
        print(f"  {msg_type}")

    result = await query_for_result(
        exporter=exporter,
        prompt="Name three primary colors. One word each, comma separated.",
        options=ClaudeAgentOptions(
            permission_mode="bypassPermissions",
            max_turns=1,
        ),
        on_message=_on_message,
    )

    print(f"\nMessage flow: {' -> '.join(message_types)}")
    print(f"Result: subtype={result.subtype}, turns={result.num_turns}")
    print("All traces exported automatically via exporter.query()")


if __name__ == "__main__":
    asyncio.run(test_wrapped_query())

File: typescript/tracing/anthropic-agents-sdk/wrapped_query_test.ts
/**
 * Wrapped Query — the simplest integration pattern.
 *
 * Uses exporter.query() which handles hooks + tracking automatically.
 * One line to instrument, zero boilerplate.
 *
 * Setup:
 *   npm install (or yarn install)
 *
 * Run:
 *   npx tsx wrapped_query_test.ts
 */

import "dotenv/config";
import { RespanAnthropicAgentsExporter } from "@respan/exporter-anthropic-agents";
import { queryForResult } from "./_sdk_runtime";

const API_KEY = process.env.RESPAN_API_KEY;
const BASE_URL = process.env.RESPAN_BASE_URL;

const exporter = new RespanAnthropicAgentsExporter({
  apiKey: API_KEY ?? undefined,
  endpoint: BASE_URL ? `${BASE_URL.replace(/\/+$/, "")}/v1/traces/ingest` : undefined,
});

async function main(): Promise<void> {
  console.log("Running wrapped query — exporter handles everything...\n");

  if (!API_KEY) {
    throw new Error("Set RESPAN_API_KEY to run this example.");
  }

  const { messageTypes, result } = await queryForResult({
    prompt: "Name three primary colors. One word each, comma separated.",
    options: exporter.withOptions({
      permissionMode: "bypassPermissions",
      maxTurns: 1,
    } as any),
    onMessage: async (message, context) => {
      await exporter.trackMessage({ message, sessionId: context.sessionId });
      console.log(`  ${String(message.type ?? "unknown")}`);
    },
  });

  console.log(`\nMessage flow: ${messageTypes.join(" -> ")}`);
  console.log(
    `Result: subtype=${String(result.subtype)}, turns=${String(result.num_turns)}`,
  );
  console.log("All traces exported automatically via exporter.query()");
}

main().catch(console.error);
